### PR TITLE
[2.19.x] DDF-5495 refactor extension for Table Export behavior

### PR DESCRIPTION
**Pull request by andrewzimmer**
#### What does this PR do?
Refactor of https://github.com/codice/ddf/pull/5496 
main goal was to avoid exporting as many impl classes and split the `executedQuery` function into a separate util the will be used as a service

#### Who is reviewing it? 
@hayleynorton @zta6 @willwill96 @cassandrabailey293 

#### Select relevant component teams: 
@codice/ui 

#### Ask 2 committers to review/merge the PR and tag them here.
@bdeining
@Lambeaux 

#### How should this be tested?
- verify no regression in exporting results from the table view

#### What are the relevant tickets?
Fixes: #5495 

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/handlers/CqlTransformHandler.java [hunk 1]
```diff
@@ -38,8 +38,9 @@
 import org.apache.tika.mime.MimeTypeException;
 import org.apache.tika.mime.MimeTypes;
 import org.codice.ddf.catalog.ui.metacard.transform.CsvTransform;
-import org.codice.ddf.catalog.ui.query.cql.CqlQueryResponse;
-import org.codice.ddf.catalog.ui.query.cql.CqlRequest;
+import org.codice.ddf.catalog.ui.query.utility.CqlQueryResponse;
+import org.codice.ddf.catalog.ui.query.utility.CqlRequest;
+import org.codice.ddf.catalog.ui.util.CqlQueryUtil;
 import org.codice.ddf.catalog.ui.util.EndpointUtil;
```
Comment (cassandrabailey293) [inline catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/handlers/CqlTransformHandler.java]: Do we still need EndpointUtil in this class?

File: catalog/ui/catalog-ui-search-api/src/main/java/org/codice/ddf/catalog/ui/query/utility/MetacardAttribute.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package org.codice.ddf.catalog.ui.query.utility;
+
```
Comment (bdeining) [inline catalog/ui/catalog-ui-search-api/src/main/java/org/codice/ddf/catalog/ui/query/utility/MetacardAttribute.java]: missing copyright headers

Comment (andrewzimmer) [reply catalog/ui/catalog-ui-search-api/src/main/java/org/codice/ddf/catalog/ui/query/utility/MetacardAttribute.java]: added

File: catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/handlers/CqlTransformHandler.java (hunk 1 above)
Comment (andrewzimmer) [reply catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/handlers/CqlTransformHandler.java]: yeah other function from endpointUtil are used https://github.com/codice/ddf/pull/5806/files/1777c3ac350e61be0512a170ec862bb5b8ba86e3#diff-cc5f97af83867437776fddd13f08d36bR124

File: catalog/ui/catalog-ui-search-api/src/main/java/org/codice/ddf/catalog/ui/query/utility/CqlQueryResponse.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.query.utility;
+
+import ddf.catalog.operation.QueryResponse;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public interface CqlQueryResponse {
```
Comment (bdeining) [inline catalog/ui/catalog-ui-search-api/src/main/java/org/codice/ddf/catalog/ui/query/utility/CqlQueryResponse.java]: we might want to mark the interfaces as experimental 
https://github.com/codice/ddf/blob/b054ac205251e31390e5ccc9830dfe310a3f2659/catalog/core/catalog-core-attachment/src/main/java/org/codice/ddf/attachment/AttachmentInfo.java#L18

**Comment by bdeining** [comment]
build now

File: catalog/ui/catalog-ui-search-api/pom.xml [hunk 4]
```diff
@@ -38,6 +38,16 @@
...
+        <dependency>
+            <groupId>org.codice.thirdparty</groupId>
+            <artifactId>geotools-suite</artifactId>
+            <version>${org.geotools.bundle.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>com.sparkjava</groupId>
+            <artifactId>spark-core</artifactId>
+            <version>${spark.version}</version>
+        </dependency>
```
Comment (Lambeaux) [inline catalog/ui/catalog-ui-search-api/pom.xml]: ❓ Are these absolutely necessary dependencies for the API? I know we use some Geotools interfaces but does that require the entire suite? Also, what Spark classes are we using in our APIs? If possible we should write the interfaces such that none of these (with the exception of the core Geotools interfaces) are used (but they're fine in the impl).

File: catalog/ui/catalog-ui-search-api/pom.xml [hunk 5]
```diff
@@ -56,7 +56,8 @@
                         <Export-Package>
                             org.codice.ddf.catalog.ui.metacard.workspace.transformer,
                             org.codice.ddf.catalog.ui.splitter,
-                            org.codice.ddf.catalog.ui.subscription
+                            org.codice.ddf.catalog.ui.subscription,
+                            org.codice.ddf.catalog.ui.query.utility
```
Comment (Lambeaux) [inline catalog/ui/catalog-ui-search-api/pom.xml]: 👍 Nice! I noticed the dependencies in this pom are gone.

File: catalog/ui/catalog-ui-search/pom.xml [hunk 6]
```diff
@@ -465,9 +465,6 @@
                         <Export-Package>
                             org.codice.ddf.catalog.ui.metacard.internal,
                             org.codice.ddf.catalog.ui.alias,
-                            org.codice.ddf.catalog.ui.api,
-                            org.codice.ddf.catalog.ui.query.cql,
-                            org.codice.ddf.catalog.ui.metacard.transform,
```
Comment (Lambeaux) [inline catalog/ui/catalog-ui-search/pom.xml]: 👍

**Comment by andrewzimmer** [comment]
build now

**Comment by cassandrabailey293** [comment]
Hero Successful 🎉 
Verified no regression in the table export functionality

**Pull request merged**